src/container/booksform.js

```diff
@@ -1,19 +1,68 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createBook } from '../actions/index';
+
 
 const category = ['Action', 'Classic', 'Drama', 'Fantasy', 'Sci-Fi', 'Legend'];
-const options = category.map(item => <option key={item}>{item}</option>);
-
-const BooksForm = () => (
-  <div>
-    <form className="input-group">
-      <input className="w-50" type="text" placeholder="Title" />
-      <select className="custom-select ml-3">
-        <option defaultValue>Choose...</option>
-        {options}
-      </select>
-      <button type="submit" className="btn btn-primary ml-2">Add Book</button>
-    </form>
-  </div>
-);
-
-export default BooksForm;
+const options = category.map(item => <option key={item} value={item}>{item}</option>);
+
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: 'Action',
+    };
+    this.handleBookChange = this.handleBookChange.bind(this);
+    this.handleBookSubmit = this.handleBookSubmit.bind(this);
+    this.handleTitleChange = this.handleTitleChange.bind(this);
+  }
+
+  handleTitleChange(e) {
+    this.setState({ title: e.target.value });
+  }
+
+
+  handleBookChange(e) {
```

Comment:
There is no need to create an event handler for each HTML element, set the name attribute on each element to be the same as the state, then in the event handler function get the element name `name = e.target.name`   now set the state of that element

Reply:
Hi Raphael, if it is just a suggestions for improvement as you said, I understand we are not required to do it? We tried to do it the way you suggest but it is not working properly in our case.